lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -30,6 +31,92 @@ def post_sql_statements( table_name, options ) # :nodoc:
     end
   end
 
+  # Add a column to be updated on duplicate key update
+  def add_column_for_on_duplicate_key_update( column, options={} ) # :nodoc:
+    arg = options[:on_duplicate_key_update]
+    if arg.is_a?( Hash )
+      columns = arg.fetch(:columns) { arg[:columns] = [] }
+      case columns
+      when Array then columns << column.to_sym unless columns.include?( column.to_sym )
+      when Hash then columns[column.to_sym] = column.to_sym
+      end
+    elsif arg.is_a?( Array )
+      arg << column.to_sym unless arg.include?( column.to_sym )
+    end
+  end
+
+  # Returns a generated ON CONFLICT DO UPDATE statement given the passed
+  # in +args+.
+  def sql_for_on_duplicate_key_update( table_name, *args ) # :nodoc:
+    arg = args.first
+    if arg.is_a?( Array ) || arg.is_a?( String )
+      arg = { :columns => arg }
+    end
+    return unless arg.is_a?( Hash )
+
+    constraint_name = arg[ :constraint_name ]
+    conflict_target = arg[ :conflict_target ]
+    columns         = arg.fetch( :columns, [] )
+
+    sql = " ON CONFLICT "
+    conflict_target = sql_for_conflict_target( conflict_target, constraint_name )
+
+    if columns.respond_to?( :empty? ) && columns.empty?
```

Comment:
Is there any test for this block? I've been trying to do something like this:
Category.import categories, :on_duplicate_key_update=>{ :conflict_target => [:name], :columns => [], validate: false }
So if the category exists I don't want to update it, just DO NOTHING